### PR TITLE
[expo-contacts][android] Fix `getContactById` never resolving when contact doesn't  exist

### LINKED ISSUE #8940: [Contacts] Contacts,getContactByIdAsync wont reject or resolve for a non existing contact id
### Summary of Issue (just a few sentences)
Contacts,getContactByIdAsync hangs when it gets a non existing id as an input.
basicaly if you want to save contacts somewhere and handle the fact that a contact does not exisit. you cant becuase the code will stop running without an error

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
was checked on android standalone and inside expo client.

### Reproducible Demo

`console.info(1);`
`await Contacts,getContactByIdAsync('fake id');`
`console.info(2);`

will never write 2 in the console.

### Steps to Reproduce
Contacts,getContactByIdAsync with a random string

### Expected Behavior vs Actual Behavior
`Contacts,getContactByIdAsync('fake id');`
expected:
return undefined or throw an error
actual:
promise will never reject or resolve; the code will hang.

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixes https://github.com/expo/expo/issues/8940.
`getContactById` never resolves when provided id doesn't belong to any contact on Android.

# How

- Resolves promise when `getContactById` returns null.
- Adds contacts module to `test-suite` on Android.

# Test Plan

- test-suite ✅

# Changelog

- Fixed `getContactById` not resolving promise when contact with provided id doesn't exist on Android.

### CODE REVIEW

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.java [hunk 1]
```diff
@@ -127,20 +128,22 @@ public void run() {
...
-          if (data == null)
-            return;
           Bundle output = new Bundle();
-          output.putParcelableArrayList("data", data);
+          if (contact != null) {
+            Collection contacts = new ArrayList();
+            contacts.add(contact);
+            ArrayList data = serializeContacts(contacts, keysToFetch, promise);
+            if (data == null)
+              return;
```
Comment (bbarthec) [inline packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.java:137]:
```suggestion
            if (data == null) {
              promise.reject(...);
              return;
            }
```
🤔 ?

**Review by bbarthec — CHANGES_REQUESTED** [review]
Spotted one code path that would prevent promise from being resolve/rejected.

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.java (hunk 1 above)
Comment (lukmccall) [reply packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.java:137]: `serializeContacts` will reject if something goes wrong 😅

**Review by bbarthec — APPROVED** [review]
LGTM 👍

**Review by tsapeta — APPROVED** [review]

**Pull request merged**